lib/webpack/plugins/formatMessage.js

```diff
@@ -0,0 +1,69 @@
+const RequestShortener = require("webpack/lib/RequestShortener");
+const requestShortener = new RequestShortener(process.cwd());
+
+function cleanStackTrace(message) {
+  return message
+    .replace(/^\s*at\s.*:\d+:\d+[\s\)]*\n/gm, ''); // at ... ...:x:y
+}
+
+function isBabelSyntaxError(e) {
+  // ModuleNotFoundError
+  // ModuleBuildError
+  return e.name === 'ModuleBuildError' && e.message.indexOf('SyntaxError') >= 0;
+}
+
+function formatMessage (webpackError) {
+
+  const error = extractError(webpackError);
+  if (isBabelSyntaxError(webpackError)) {
+    error.message = cleanStackTrace(error.message + '\n');
+  }
+
+  return error;
+}
+
+function extractError (e) {
```

Comment:
@gaearon Totally! I was thinking about extracting a module for this plugin. 
It still needs a bit of work and some tests.
I'll make a PR when it's ready 😄